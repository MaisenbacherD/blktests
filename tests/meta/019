#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2024 Western Digital Corporation or its affiliates.
#
# Test _set_combined_conditions() helper

. tests/meta/rc

DESCRIPTION="combine three set_conditions() hooks"

conditions_x() {
	local index=$1

	if [[ -z $index ]]; then
		echo 2
		return
	fi

	meta_x_index=$index
	COND_DESC="x=$index"
}

conditions_y() {
	local index=$1

	if [[ -z $index ]]; then
		echo 2
		return
	fi

	meta_y_index=$index
	COND_DESC="y=$index"
}

conditions_z() {
	local index=$1

	if [[ -z $index ]]; then
		echo 2
		return
	fi

	meta_z_index=$index
	COND_DESC="z=$index"
}

set_conditions() {
	_set_combined_conditions conditions_x conditions_y conditions_z "$@"
}

test() {
	echo "Running ${TEST_NAME}"
	echo "(x,y,z)=(${meta_x_index},${meta_y_index},${meta_z_index})" >> "$FULL"
	echo "Test complete"
}
