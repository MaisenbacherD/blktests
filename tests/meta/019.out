Running meta/019
Test complete
